controller/volume_controller.go

```diff
@@ -3005,6 +3006,146 @@ func (vc *VolumeController) ResolveRefAndEnqueue(namespace string, ref *metav1.O
 	vc.enqueueVolume(volume)
 }
 
+// createNewRecurringJobNameUsed use a new name to create a new recurring job that original name exists but configuration is different
+func createNewRecurringJobNameUsed(log *logrus.Entry, ds *datastore.DataStore, job *longhorn.RecordVolumeRecurringJob) (string, error) {
+	// reuse restored recurring job that RecurringJobSpec is the same.
+	allRecurringJobs, err := ds.ListRecurringJobsRO()
+	if err != nil {
+		log.WithError(err).Warn("Failed to list all recurring jobs")
+	}
+	for existJobName, existJob := range allRecurringJobs {
+		if !strings.HasPrefix(existJobName, types.VolumeRecurringJobRestorePrefix) {
+			continue
+		}
+		job.JobSpec.Name = existJob.Spec.Name
+		if reflect.DeepEqual(existJob.Spec, job.JobSpec) {
+			return existJobName, nil
+		}
+	}
+
+	newJobName := types.VolumeRecurringJobRestorePrefix + util.RandomID() + util.RandomID()
```

Comment:
nit: add a helper function for it
```suggestion
	newJobName := generateJobName(types.VolumeRecurringJobRestorePrefix)
```

Reply:
Added. Thanks